### DISCUSSION TITLE
Video Output and Bitrate Control in openh264 Encoder

### DISCUSSION THREAD

**Discussion by USAMAWIZARD** [ant-media/Ant-Media-Server#5386] [FAQs] [answered]
What does the below log message,
`INFO  i.a.e.adaptive.video.H264Encoder - There is no video output at this time for stream index 0 for stream:abcd. Encoder may be dropping to control the bitrate. You can either increase bitrate or you can change rate control to quality (settings.encoding.rc=quality) for openh264`
indicate in the context of openh264 video encoding?

Thank you

**Comment by Mohit-3196** [comment, marked as answer]
The provided log message,
`INFO i.a.e.adaptive.video.H264Encoder - There is no video output at this time for stream index 0 for stream:abcd. Encoder may be dropping to control the bitrate. You can either increase bitrate or you can change rate control to quality (settings.encoding.rc=quality) for openh264,`
is related to the openh264 video encoder.
This message indicates that the encoder is experiencing a lack of video output for a specific stream index ("stream:abcd"), which may be caused by the encoder dropping to control the bitrate when adaptive setting is enabled.

To address this issue, as suggested in the log messages there are two potential solutions:

1. Increase the bitrate to provide a sufficient data rate for encoding the video stream.
2. Change the rate control to the quality mode by modifying the parameter `settings.encoding.rc=quality` in the related application settings.

Cheers.
